src/Symfony/Component/PropertyInfo/Util/PhpDocTypeHelper.php

```diff
@@ -115,43 +115,29 @@ private function createType(DocType $type, bool $nullable, string $docType = nul
 
             [$phpType, $class] = $this->getPhpTypeAndClass((string) $fqsen);
 
-            $key = $this->getTypes($type->getKeyType());
-            $value = $this->getTypes($type->getValueType());
-
-            // More than 1 type returned means it is a Compound type, which is
-            // not handled by Type, so better use a null value.
-            $key = 1 === \count($key) ? $key[0] : null;
-            $value = 1 === \count($value) ? $value[0] : null;
-
-            return new Type($phpType, $nullable, $class, true, $key, $value);
+            $keys = $this->getTypes($type->getKeyType());
```

Comment:
if this returns an empty array, we should still use `null` to mark collection keys as unknown

Reply:
Should i change it, even though it will be converted to empty array in `Type::__constructor`?

Reply:
ah, I forgot that Type makes no difference there. So this is indeed not needed